applications/StructuralMechanicsApplication/python_scripts/read_materials_process_with_create.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Other imports
+import sys
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ReadMaterialsProcess(Model, settings["Parameters"])
+
+
+class ReadMaterialsProcess(KratosMultiphysics.Process):
```

Comment:
I do not agree with copy pasting things from the core and modifying those, this affects everyone using StructuralMechanics!

Such things should be discussed in a separate PR

Is this included in #2291 ?

Reply:
Nope, this should be discussed in parallel indeed